connectors/sources/github_graphql.py

```diff
@@ -0,0 +1,889 @@
+            self.github_client.user = await self.github_client.get_logged_in_user()
+
+            for repo_name in self.github_client.repos:
+                if repo_name not in ["", None]:
+                    if "/" in repo_name:
+                        foreign_repos.append(repo_name)
+                    else:
+                        configured_repos.append(
+                            f"{self.github_client.user}/{repo_name}"
+                        )
+            async for repo in self.github_client.get_user_repos():
+                self.user_repos[repo["nameWithOwner"]] = repo
+            invalid_repos = list(set(configured_repos) - set(self.user_repos.keys()))
+
+            for repo_name in foreign_repos:
+                try:
+                    self.foreign_repos[
+                        repo_name
+                    ] = await self.github_client.get_foreign_repo(repo_name=repo_name)
+                except Exception:
+                    invalid_repos.append(repo_name)
+            return invalid_repos
+        except Exception as exception:
+            self._logger.exception(
+                f"Error while checking for inaccessible repositories. Exception: {exception}."
+            )
+            raise
+
+    async def _remote_validation(self):
+        """Validate scope of the configured Token and accessibility of repositories
+
+        Raises:
+            ConfigurableFieldValueError: Insufficient privileges error.
+        """
+        query_data = {"query": GithubQuery.USER_QUERY.value, "variables": None}
+        _, headers = await self.github_client.api_call(  # pyright: ignore
+            query_data=query_data, need_headers=True
+        )
+        if "repo" not in headers.get("X-OAuth-Scopes"):  # pyright: ignore
+            raise ConfigurableFieldValueError(
+                "Configured token does not have required rights to fetch the content"
+            )
+        if self.github_client.repos != [WILDCARD]:
+            invalid_repos = await self.get_invalid_repos()
+            if invalid_repos:
+                raise ConfigurableFieldValueError(
+                    f"Inaccessible repositories '{', '.join(invalid_repos)}'."
+                )
+
+    async def validate_config(self):
+        """Validates whether user input is empty or not for configuration fields
+        Also validate, if user configured repositories are accessible or not and scope of the token
+        """
+        self.configuration.check_valid()
+        await self._remote_validation()
+
+    async def close(self):
+        await self.github_client.close()
+
+    async def ping(self):
+        try:
+            await self.github_client.ping()
+            self._logger.debug("Successfully connected to GitHub.")
+        except Exception:
+            self._logger.exception("Error while connecting to GitHub.")
+            raise
+
+    def adapt_gh_doc_to_es_doc(self, github_document, schema):
+        document = {}
```

Comment:
nit: that could be done using a dict comprehension. Feel free to leave it like this.